File: LivrusVirtus/VCPP6/wxWin/LibFireBird/ConPaises.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConPaises.h"

BEGIN_EVENT_TABLE(CfrmConPaises, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConPaises::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConPaises::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConPaises::OnBtnFechar)
  EVT_CLOSE(CfrmConPaises::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConPaises::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConPaises::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConPaises::CfrmConPaises(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConPaises::~CfrmConPaises() { }

void CfrmConPaises::IniciaControles() {
 this->SetTitle(STR_CONPAIS);
 this->labDescricao->SetLabel(wxT("País:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("País"));
}

void CfrmConPaises::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_PAISES);
  event.Skip();
}

void CfrmConPaises::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConPaises::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConPaises::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Pais('S',wxT(""), edtDescricao->GetValue()), STR_PAISNENC, 2);
  event.Skip();
}

void CfrmConPaises::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConPaises::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

File: LivrusVirtus/wxWindows/wxRuby/ConPaises.rb
require 'wxruby'
require 'Consultas'
require 'resource'
require 'RotinasGlobais'
require 'ConsultasSQL'

class FrmConPaises < FrmConsultas
  def initialize(parent)
    super(parent) 
    set_title(Resource::STR_CONPAIS)
    @labDescricao.set_label('Pais:')
    @gridConsulta.set_col_label_value(1, "Pais")

    evt_button(ID_FRMCONSULTAS_BTNPESQUISAR) {|event| OnBtnPesquisarClick(event)}
    evt_button(ID_FRMCONSULTAS_BTNLIMPAR) {|event| OnBtnLimparClick(event)}
    evt_button(ID_FRMCONSULTAS_BTNFECHAR) {|event| OnBtnFecharClick(event)}
    evt_close {|event| on_close(event)}
    evt_grid_select_cell {|event| OnGridSelectCell(event)}
    evt_grid_label_left_click {|event| OnGridLabelLeftClick(event)}
  end

  def on_close(event)
    self.Fecha(Resource::MNU_CON_PAISES)
    event.skip()
  end

  def OnBtnPesquisarClick(event)
    self.PesquisarDados(
      $ConsSQL.Pais('S', "", @edtDescricao.get_value()),
      Resource::STR_PAISNENC, 2)
    event.skip()
  end

  def OnBtnLimparClick(event)
    self.LimparDados()
    event.skip()
  end

  def OnBtnFecharClick(event)
    self.Fecha(0)
    event.skip()
  end

  def OnGridSelectCell(event)
    self.setLabRegistros(event.get_row())
    event.skip()
  end
  
  def OnGridLabelLeftClick(event)
    self.setLabRegistros(event.get_row())
    event.skip()
  end

end

